hive-metastore/src/main/java/org/apache/iceberg/hive/HiveTableOperations.java

```diff
@@ -90,6 +92,7 @@
   private static final String HIVE_LOCK_CHECK_MAX_WAIT_MS = "iceberg.hive.lock-check-max-wait-ms";
   private static final String HIVE_ICEBERG_METADATA_REFRESH_MAX_RETRIES = "iceberg.hive.metadata-refresh-max-retries";
   private static final String HIVE_TABLE_LEVEL_LOCK_EVICT_MS = "iceberg.hive.table-level-lock-evict-ms";
+  private static final long HIVE_TABLE_PROPERTY_VALUE_SIZE_MAX = 4000;
```

Comment:
Should we should expose this configuration, otherwise it may not match what works for all users (different database and backends)?  I am not sure what others think.

Reply:
Ideally, it should be configurable. Not sure it's worth to do that at this moment. I'm open to any suggestion.

Reply:
I guess my comment is not complete, @flyrain's analysis in https://github.com/apache/iceberg/pull/4456#discussion_r840899312 is the exact one.  Older versions of Hive had the 4000 limit (due to need to support some older databases), and newer versions increased to 32672.  

My point was rather, if we don't have some kind of flag, we lose the full potential of current HMS versions just to support the older ones.

Reply:
Hm, yea I'm also not entirely sure what the right thing to do here is.  Yea it's hidden a bit deeply in the [package.jdo](https://github.com/apache/hive/blob/master/standalone-metastore/metastore-server/src/main/resources/package.jdo) file in Hive..

I see we do have some Hive specific configs in this file right (like iceberg.hive.lock-timeout-ms for example), and also ConfigProperties.java?  Could it be with that prefix (ie, iceberg.hive.summary.stats.max.size), with default value  and upper bound 32672?   Or if we don't want to give too much complexity to user, maybe  iceberg.hive.legacy.table.parameter.size which is just a boolean controlling whether its 32672 or 4000.

Reply:
Sounds good. The only downside is we cannot separate them when two hive catalogs are configured, but that's probably fine. `iceberg.hive.lock-timeout-ms` couldn't either.